spectrapepper/functions.py

```diff
@@ -2806,3 +2806,45 @@ def asymmetry(data, peak, axis, res= 0.001, s=5, limit=10):
         diff_abs = np.sqrt(diff_abs)*(-1)
     
     return diff_abs
+
+def running_median_nd(data, window_size,plot=False):
+    """
+    Computes the median of an array in a running window defined by the user.
+    The window size needs to be a 2D tuple or list, with the first element being the
+    length of the spectra and the second the total width that will be taken into account
+    for the statistics.
+
+    :type data: numpy array
+    :param data: The spectras 
+
+    :type window_size: tuple/list
+    :param window_size: Window size parameters
+
+    :type plot: boolean
+    :param plot: If you want to plot the new spectra change to True
+
+    :returns: Array containing the computed 1D spectra.
+    :rtype: numpy array[float]
+    """
+    # Check dimensions
+    if np.shape(window_size)[0] != 2:
+        raise ValueError("The window must be 2D")
+    if (window_size[1] % 2) == 0:
```

Comment:
seems arbitrary restriction, median is perfectly well defined for even number of elements

Reply:
The idea behind that restriction was to keep the spectra from shifting left or right after the processing